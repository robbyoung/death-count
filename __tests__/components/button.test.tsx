import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../app/components/button';

describe('Button Component', () => {
    it('renders correctly', () => {
        const component = renderer.create(<Button
            text="Hello Button"
            onPress={() => undefined}
        />);
        expect(component.toJSON()).toMatchSnapshot();
    });
});