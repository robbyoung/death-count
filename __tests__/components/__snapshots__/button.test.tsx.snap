// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component renders correctly 1`] = `
<View
  accessible={true}
  clickable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "marginTop": 10,
      "opacity": 1,
      "width": "100%",
    }
  }
>
  <Text
    style={
      Object {
        "backgroundColor": "#e84641",
        "borderRadius": 5,
        "color": "#ffffff",
        "fontSize": 30,
        "height": 50,
        "marginLeft": "5%",
        "marginRight": "5%",
        "textAlign": "center",
        "textAlignVertical": "center",
        "width": "90%",
      }
    }
  >
    Hello Button
  </Text>
</View>
`;
